### PR TITLE
Geocoding service

### PR DESCRIPTION
**Pull request by davidtheclark**
The basics are in place, but this PR is dependent on @kepta's pending changes to the validator.

### CODE REVIEW

**Comment by kepta** [comment]
just updated the validator part.

**Comment by davidtheclark** [comment]
@kepta This is now ready for review.

File: services/geocoding.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+ * @param {string|Array<string>} [config.language]
+ * @return {MapiRequest}
+ */
+Geocoding.forwardGeocode = function(config) {
+  v.assertShape({
+    query: v.required(v.string),
+    mode: v.required(v.oneOf('mapbox.places', 'mapbox.places-permanent')),
+    country: v.oneOfType(v.string, v.arrayOf(v.string)),
+    proximity: v.coordinates,
+    types: v.arrayOf(v.string),
```
Comment (kepta) [inline services/geocoding.js]: We could check for all options here `Options are  country ,  region ,  postcode ,  district , place ,  locality ,  neighborhood ,  address ,  poi , and  poi.landmark .`

**Comment by davidtheclark** [comment]
@kepta I added `oneOf` validation for the `type` options. In order to avoid repeating all those strings I used an array as the argument to `oneOf()`, which meant a little change to `validator.js`. Given this use case that makes arrays useful, I wonder if we should consider expecting arrays always instead of allow either argument lists *or* arrays. I know you're often interested in cutting down API surface 😄 What do you think?

**Comment by davidtheclark** [comment]
I'm going to merge this PR, and we can consider changes to the validator as follow-up tasks.

**Pull request merged**